cli/azd/pkg/devcenter/platform.go

```diff
@@ -165,13 +165,13 @@ func (p *Platform) ConfigureContainer(container *ioc.NestedContainer) error {
 
 	// Other devcenter components
 	container.MustRegisterSingleton(func(
-		ctx context.Context,
 		credential azcore.TokenCredential,
 		httpClient httputil.HttpClient,
 		resourceGraphClient *armresourcegraph.Client,
 	) (devcentersdk.DevCenterClient, error) {
-		options := azsdk.
-			DefaultClientOptionsBuilder(ctx, httpClient, "azd").
+		options := azsdk.NewClientOptionsBuilderFactory(httpClient, "azd").
```

Comment:
Does the user agent here need to be "azd" or can we use the normal user agent logic?

Reply:
This can be normal azd user agent.

Reply:
Looks like user agent consistency is discussed in another issue... https://github.com/Azure/azure-dev/issues/2723 

That can be addressed separately.